app/server.go

```diff
@@ -318,41 +324,33 @@ func tryConnect(ip, binding network.Address) error {
 	log.Info("Succesfully connected istelf to port")
 	conn.Close()
 
-	_, port, err := net.SplitHostPort(ip.NetworkAddress())
+	_, portStr, err := net.SplitHostPort(ip.NetworkAddress())
 	if err != nil {
 		return err
 	}
-	values := url.Values{}
-	values.Set("port", port)
-	values.Set("timeout", "default")
 
-	// ask the check
-	url := whatsMyIP + "port-scanner/scan.php"
-	req, err := http.NewRequest("POST", url, bytes.NewBufferString(values.Encode()))
+	port, err := strconv.Atoi(portStr)
 	if err != nil {
 		return err
 	}
-	req.Header.Set("Host", "www.whatsmyip.org")
-	req.Header.Set("Referer", "http://www.whatsmyip.org/port-scanner/")
-	req.Header.Set("User-Agent", "Mozilla/5.0 (X11; Linux x86_64; rv:46.0) Gecko/20100101 Firefox/46.0")
-	req.Header.Set("Accept", "*/*")
-	req.Header.Set("Accept-Language", "en-US,en;q=0.5")
-	req.Header.Set("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
-	req.Header.Set("X-Requested-With", "XMLHttpRequest")
-
-	client := &http.Client{}
-	resp, err := client.Do(req)
+
+	// ask the check
+	url := fmt.Sprintf("%s?port=%d", portscan, port)
+	resp, err := http.Get(url)
+	// can't get the public ip then ask the user for a reachable one
 	if err != nil {
-		return err
+		return errors.New("Could not get your public IP address")
 	}
 
-	buffer, err := ioutil.ReadAll(resp.Body)
+	buff, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
 	if err != nil {
 		return err
 	}
 
-	if !bytes.Contains(buffer, []byte("1")) {
-		return errors.New("Address unreachable")
+	res := string(buff)
+	if strings.Compare("Open", res) != 0 {
```

Comment:
It's nitpicking but it seems overly complicated for a string comparison... why not simply `res == "Open"` ?